### PR TITLE
Bring o-date more inline with other Origami components.

### PR DESCRIPTION
**Pull request by notlee**
- Move o-date logic out of `main.js`, keeping shared
`o.DOMContentLoaded` event initialisation logic.
- Use `class` syntax.
- Deprecate non-typical constructor options in favour of using
the `init` method.
- Add further deprecation warnings for `ft-date-format` methods.

### CODE REVIEW

File: src/js/date.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    constructor(rootEl) {
+
+        if (!rootEl) {
+            console.warn('To initalise all o-date elements on the page use ' +
+                'the `init` method. Passing no arguments to the constructor ' +
+                'is deprecated.');
+        }
+
+        if (rootEl && !(rootEl instanceof HTMLElement)) {
+            console.warn('Using the constructor to initalise one of more ' +
```
Comment (JakeChampion) [inline src/js/date.js:25]:
```suggestion
            console.warn('Using the constructor to initialise one or more ' +
```

File: src/js/date.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+			}, 60000);
+		}
+	}
+
+	// Use object-level event listener method so a new function doesn't need to be bound for each instance
+	handleEvent(e) {
+		if (e.type === updateEventName) {
+			this.update();
+		}
+	};
```
Comment (JakeChampion) [inline src/js/date.js:61]:
```suggestion
	}
```

File: src/js/date.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+		if (dateString) {
+			el.setAttribute('title', ftDateFormat.format(date, 'datetime'));
+			el.setAttribute('aria-label', labelString || dateString);
+			el.removeAttribute('aria-hidden');
+		} else {
+			el.removeAttribute('title');
+			el.removeAttribute('aria-label');
+			el.setAttribute('aria-hidden', '');
+		}
+	};
```
Comment (JakeChampion) [inline src/js/date.js:128]:
```suggestion
	}
```

File: src/js/date.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+	/**
+	 * Remove o-date from the `time` element i.e. remove event
+	 * listeners and drop references to the element.
+	 */
+	destroy() {
+		document.body.removeEventListener(updateEventName, this);
+		this.el = null;
+	};
```
Comment (JakeChampion) [inline src/js/date.js:137]:
```suggestion
	}
```

File: src/js/date.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+		if (/\bo-date\b/.test(el.getAttribute('data-o-component'))) {
+			return new ODate(el);
+		}
+
+		// If el contains date components, return o-dates
+		const dateEls = el.querySelectorAll('[data-o-component~="o-date"]');
+		return [].map.call(dateEls, function (el) {
+			return new ODate(el);
+		});
+	};
```
Comment (JakeChampion) [inline src/js/date.js:162]:
```suggestion
	}
```

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 4.0.8 to e225203018e170a8242ab0fe4e9f49aa53da3f92
js: 2.29kb increase (0.59kb/gzip)

**Comment by notlee** [comment]
thanks for the linting fixes @JakeChampion !

**Comment by notlee** [comment]
As a new feature is incoming I'll merge this but hold the release

**Comment by JakeChampion** [comment]
> thanks for the linting fixes @JakeChampion !

I needed a short o-tracking break 😅

**Pull request merged**